experiment/simulation/js/fa.js

```diff
@@ -0,0 +1,333 @@
+import { registerGate } from "./main.js";
+import { setPosition } from "./layout.js";
+import {gates} from './gate.js';
+import {jsPlumbInstance} from "./main.js";
+
+// Dictionary of all full adders in the circuit with their IDs as keys
+let fullAdder = {}
+
+function clearFAs() {
+
+    for (let faID in fullAdder) {
+        delete fullAdder[faID];
+    }
+    fullAdder = {};
+}
+
+// {output-id: [gate,pos]}
+let finalOutputs = {
+    "Output-5": [],
+    "Output-8": [],
+    "Output-11": [],
+    "Output-12": [],
+};
+
+class FullAdder {
+    constructor() {
+        this.id = "FullAdder-" + window.numComponents++;
+        this.a0 = [];  // Takes 2 items in a list : Gate, Output endpoint of gate
+        this.b0 = [];
+        this.Cin = [];
+        this.sum = null;
+        this.Cout = null;
+        this.inputPoints = [];
+        this.outputPoints = [];
+        this.CoutIsConnected = false;
+        this.sumIsConnected = false;
+        this.component = '<div class="drag-drop FullAdder" id=' + this.id + ' style="width:100px;height:100px;"></div>';

```

Comment:
Please use [template literals](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals) instead of string concatenation for better performance.

Reply:
done as instructed